src/widgets/results_widgets/k_means_results.py

```diff
@@ -132,3 +138,80 @@ def update_plot(self):
+
+            self.centroids_canvas.chosen_centroid_plot(x, y, None, None, x_centroids.iloc[row], y_centroids.iloc[row],
+                                                       row, len(x_centroids), self.ox, self.oy, min_x - sep_x,
+                                                       max_x + sep_x, min_y - sep_y, max_y + sep_y,)
+        else:
+            x = self.data[self.ox]
+            y = self.data[self.oy]
+            min_x = x.min()
+            max_x = x.max()
+            min_y = y.min()
+            max_y = y.max()
+            sep_x = 0.1 * (max_x - min_x)
+            sep_y = 0.1 * (max_y - min_y)
+
+            x_centroids = self.centroids[self.ox]
+            y_centroids = self.centroids[self.oy]
+
+            self.centroids_canvas.new_centroids_plot(None, None, x_centroids, y_centroids, self.ox, self.oy,
+                                                     min_x - sep_x, max_x + sep_x, min_y - sep_y, max_y + sep_y,
+                                                     )
+
+    def show_cluster(self):
+        cluster_num = 0
+        for idx in self.centroids_table.selectionModel().selectedIndexes():
+            cluster_num = idx.row()
+        rows = [i for i in range(len(self.labels)) if self.labels[i] == cluster_num]
+        elements = self.data.iloc[rows]
+        self.centroids_table.setModel(QtTable(elements))
+        buttons_widget = QWidget()
+        buttons_layout = QHBoxLayout()
+        exit_button = QPushButton("X")
+        exit_button.clicked.connect(self.exit_from_cluster)
+        exit_button.setFixedWidth(50)
+        save_button = QPushButton("SAVE")
+        save_button.clicked.connect(partial(self.on_save_button_click, elements))
+        save_button.setFixedWidth(100)
+        buttons_layout.addWidget(save_button)
+        buttons_layout.addWidget(exit_button)
+        buttons_layout.setAlignment(Qt.AlignRight)
+        buttons_widget.setLayout(buttons_layout)
+        self.centroids_group_layout.insertWidget(0, buttons_widget)
+        self.centroids_table_instruction.hide()
+        self.centroids_table.doubleClicked.disconnect()
+        self.update_cluster_plot(cluster_num)
+
+    def exit_from_cluster(self):
+        self.centroids_table.setModel(QtTable(self.centroids.round(3)))
+        self.centroids_group_layout.itemAt(0).widget().setParent(None)
+        self.centroids_table_instruction.show()
+        self.centroids_table.doubleClicked.connect(self.show_cluster)
+        self.update_cluster_plot()
+
+    def on_save_button_click(self, elements):
+        path, is_ok = QInputDialog.getText(self, 'input name', 'Enter filename')
+        if is_ok and path and path.endswith(".csv"):
+            elements.to_csv(path)
+        else:
+            error = QMessageBox()
+            error.setIcon(QMessageBox.Critical)
+            error.setText("An error has occurred while trying to write cluster to file")
+            error.setWindowTitle("Error")
+            error.exec_()
```

Comment:
We shouldn't show an error when the user simply cancelled the window. I assume is_ok is False in that case. Also when the extension is wrong the error message should state that the file type is incorrect. The current message makes sense if there was indeed an exception while trying to create/access the file.